runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinClasspathUtils.kt

```diff
@@ -41,12 +48,20 @@ private fun KotlinCompilation.compileClasspathOf(project: Project): FileCollecti
         return this.classpathOf(project)
     }
 
-    val platformDependencyFiles: FileCollection = (this as? AbstractKotlinNativeCompilation)
-        ?.target?.project?.configurations
-        ?.findByName(@Suppress("DEPRECATION") this.defaultSourceSet.implementationMetadataConfigurationName) // KT-58640
-        ?: project.files()
+    val kgpVersion = project.getKgpVersion()
+    val shouldKeepBackwardsCompatibility = (kgpVersion != null && kgpVersion < KotlinGradlePluginVersion(1, 8, 0))
+
+    return if (shouldKeepBackwardsCompatibility) {
+        val platformDependencyFiles: FileCollection = (this as? AbstractKotlinNativeCompilation)
+            ?.target?.project?.configurations
+            ?.findByName(@Suppress("DEPRECATION") this.defaultSourceSet.implementationMetadataConfigurationName) // KT-58640
+            ?: project.files()
 
-    return this.compileDependencyFiles + platformDependencyFiles + this.classpathOf(project)
+        this.compileDependencyFiles + platformDependencyFiles + this.classpathOf(project)
+    } else {
+        // compileTaskProvider is available starting from kotlin 1.8.0
+        (compileTaskProvider.get() as? KotlinCompileTool)?.run { sources + libraries } ?: project.files()
```

Comment:
I ended up with the following  changes that work:

```suggestion
        this.compileDependencyFiles + platformDependencyFiles + this.classpathOf(project)
    } else {
        this.compileDependencyFiles + this.classpathOf(project)
```

`(compileTaskProvider.get() as? KotlinCompileTool)` is already returned by `getKotlinCompileTask` (called transitively through `classpathOf`), and the `libraries` files are also returned by the `classpathOf` overload below, so this bit can be simplified.

Removing `this.compileDependencyFiles` feels scary as I don't know what it was introduced for, and I'd personally leave it if it doesn't get in the way.

Calling `sources` doesn't seem to make any difference, but I haven't tested it extensively - was there a particular reason you used it? I mean the `run { sources + libraries }` bit

Reply:
> (compileTaskProvider.get() as? KotlinCompileTool) is already returned by getKotlinCompileTask

I think, it's not, because in `getKotlinCompileTask` task is casted to `KotlinCompile`, which AFAIK is JVM only, but in PR `KotlinCompileTool` is used - which is implemented by all platform compilation tasks (JVM/Native/JS). Still, may be it's enough to cast to `KotlinCompileTool` in `getKotlinCompileTask`, but Im not sure about the specific version when this interface was introducec

> Removing this.compileDependencyFiles feels scary

While Im not 100% sure, but recently introduced `libraries` should contain everything what should exists in `compileDependencyFiles` (as from what I see while diving in KGP source code)
The main reason for not to use it, as it's mutable variable, which could be not fully configured (for some unknown reason) when fetching it - I don't know if it's true or not. I would go with using just `libraries` (at least starting from 1.9.0), as it's looks like real public API, comparing to `compileDependencyFiles` + introducing fallback property (on task, or just some gradle property) in Dokka plugin to also add `compileDependencyFiles` - this is just because Im afraid that if using it, may be in next KGP version something will change again

> Calling sources doesn't seem to make any difference

Yeah, I think it could be removed, as Dokka process sources separately in `KotlinSourceSetGist`

Reply:
> I think, it's not, because in getKotlinCompileTask task is casted to KotlinCompile, which AFAIK is JVM only, but in PR KotlinCompileTool is used - which is implemented by all platform compilation tasks (JVM/Native/JS)

Oh, I didn't even think in that direction! I checked that `KotlinCompile` implements `KotlinCompileTool`, but judging by the name alone I didn't expect it to be JVM-only... 

Thank you for explaining your reasoning, it makes total sense, and it looks like it should be safe to proceed like you propose. Especially if we leave a gradle property to revert to the old behaviour with `platformDependencyFiles` and `compileDependencyFiles ` 👍 

Just in case, I'll discuss this and other matters with a Multiplatform/KGP expert tomorrow morning, in case there's something we're missing, and get back to you